packages/url_launcher/url_launcher_android/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncher.java

```diff
@@ -15,99 +15,122 @@
-   * @param useWebView when true, the URL is launched inside of {@link WebViewActivity}.
-   * @param enableJavaScript Only used if {@param useWebView} is true. Enables JS in the WebView.
-   * @param enableDomStorage Only used if {@param useWebView} is true. Enables DOM storage in the
-   * @return {@link LaunchStatus#NO_ACTIVITY} if there's no available {@code applicationContext}.
-   *     {@link LaunchStatus#ACTIVITY_NOT_FOUND} if there's no activity found to handle {@code
-   *     launchIntent}. {@link LaunchStatus#OK} otherwise.
-   */
-  LaunchStatus launch(
-      @NonNull String url,
-      @NonNull Bundle headersBundle,
-      boolean useWebView,
-      boolean enableJavaScript,
-      boolean enableDomStorage) {
-    if (activity == null) {
-      return LaunchStatus.NO_ACTIVITY;
-    }
+  @Override
+  public @NonNull Boolean launchUrl(@NonNull String url, @NonNull Map<String, String> headers) {
+    ensureActivity();
+    assert activity != null;
 
-    Intent launchIntent;
-    if (useWebView) {
-      launchIntent =
-          WebViewActivity.createIntent(
-              activity, url, enableJavaScript, enableDomStorage, headersBundle);
-    } else {
-      launchIntent =
-          new Intent(Intent.ACTION_VIEW)
-              .setData(Uri.parse(url))
-              .putExtra(Browser.EXTRA_HEADERS, headersBundle);
+    Intent launchIntent =
+        new Intent(Intent.ACTION_VIEW)
+            .setData(Uri.parse(url))
+            .putExtra(Browser.EXTRA_HEADERS, extractBundle(headers));
+    try {
+      activity.startActivity(launchIntent);
+    } catch (ActivityNotFoundException e) {
+      return false;
     }
 
+    return true;
+  }
+
+  @Override
+  public @NonNull Boolean openUrlInWebView(@NonNull String url, @NonNull WebViewOptions options) {
+    ensureActivity();
+    assert activity != null;
+
+    Intent launchIntent =
+        WebViewActivity.createIntent(
+            activity,
+            url,
+            options.getEnableJavaScript(),
+            options.getEnableDomStorage(),
+            extractBundle(options.getHeaders()));
     try {
       activity.startActivity(launchIntent);
     } catch (ActivityNotFoundException e) {
-      return LaunchStatus.ACTIVITY_NOT_FOUND;
+      return false;
     }
 
-    return LaunchStatus.OK;
+    return true;
   }
 
-  /** Closes any activities started with {@link #launch} {@code useWebView=true}. */
-  void closeWebView() {
+  @Override
+  public void closeWebView() {
     applicationContext.sendBroadcast(new Intent(WebViewActivity.ACTION_CLOSE));
   }
 
-  /** Result of a {@link #launch} call. */
-  enum LaunchStatus {
-    /** The intent was well formed. */
-    OK,
-    /** No activity was found to launch. */
-    NO_ACTIVITY,
-    /** No Activity found that can handle given intent. */
-    ACTIVITY_NOT_FOUND,
+  private static @NonNull Bundle extractBundle(Map<String, String> headersMap) {
```

Comment:
This, and the strings in the error in `ensureActivity`, are the surviving code from `MethodCallHandlerImpl`.